authors/apps/article_tag/views.py

```diff
@@ -0,0 +1,21 @@
+from django.shortcuts import render
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework.viewsets import ModelViewSet
+from authors.apps.article_tag.models import ArticleTag
+from authors.apps.article_tag.serializers import ArticleTagSerializer
+
+class ArticleTagViewSet(ModelViewSet):
+    queryset = ArticleTag.objects.all()
+    serializer_class = ArticleTagSerializer
+
+
+    def create_tag_if_not_exist(self, provided_tags):
+        """method to create a tag if it doesn't exist"""
+        for provided_tag in provided_tags:
```

Comment:
@well done @llwasampijja, I have tried out your endpoint and our previous error of tag does not exist on creating an article has been resolved. However, I feel that we can improve the functionality for creating the tag if it does not exist by first creating a unique set of tags provided by the user so that the loop `for provided_tag in provided_tags:` does not run for duplicate tags
```
unique_tags = {tag for tag in provided_tags}
for tag in unique_tags:
```
If the user provides tags like `["React", "Python", "React", "Python"]
the loop will run 2 times using the unique_tags whereas if we use provided_tags it will run three times. As the duplicates grow bigger, they will have an impact both on time and space complexity of our product.
Also, consider converting the tags to lower case when comparing. so that if the user provides a list like ```["python', "Python"]```, we do not create two tags 
-  Make the case of the tags insensitive
- Also, remove duplicate tags with a set compression

Reply:
@kalsmic I have improved it!

Reply:
well done @llwasampijja, it is now functioning as expected. kudos